include/pybind11/cast.h

```diff
@@ -377,6 +377,21 @@ template <typename StringType, bool IsView = false> struct string_caster {
 #endif
         }
 
+#if PY_VERSION_HEX >= 0x03030000
+        // On Python >= 3.3, for UTF-8 we avoid the need for a temporary `bytes`
+        // object by using `PyUnicode_AsUTF8AndSize`.
+        if (UTF_N == 8) {
+            Py_ssize_t size;
+            const CharT *buffer
```

Comment:
```suggestion
            Py_ssize_t size = 0;
            const auto *buffer
```

Reply:
Changed to use `auto`.  Since it is not actually necessary to initialize `size`, I added `NOLINT` instead to silence clang-tidy.